modules/cli/src/main/scala/scala/cli/exportCmd/JsonProject.scala

```diff
@@ -0,0 +1,92 @@
+package scala.cli.exportCmd
+
+import com.github.plokhotnyuk.jsoniter_scala.core.{JsonValueCodec, WriterConfig, writeToStream}
+import com.github.plokhotnyuk.jsoniter_scala.macros.{CodecMakerConfig, JsonCodecMaker}
+import coursier.Dependency
+import coursier.util.Artifact
+
+import java.nio.charset.StandardCharsets
+
+import scala.build.options.ConfigMonoid
+import scala.cli.util.SeqHelpers.*
+import scala.reflect.NameTransformer
+import scala.util.{Properties, Using}
+
+final case class JsonProject(
+  projectName: Option[String] = None,
+  scalaVersion: Option[String] = None,
+  platform: Option[String] = None,
+  scalacOptions: Seq[String] = Nil,
+  scalaCompilerPlugins: Seq[String] = Nil,
+  scalaJsVersion: Option[String] = None,
+  scalaNativeVersion: Option[String] = None,
+  mainDeps: Seq[ExportDependencyFormat] = Nil,
+  testDeps: Seq[ExportDependencyFormat] = Nil,
+  mainSources: Seq[String] = Nil,
+  testSources: Seq[String] = Nil,
+  resolvers: Seq[ExportResolverFormat] = Nil,
+  extraDecls: Seq[String] = Nil,
+  resourcesDirs: Seq[String] = Nil,
+  mainClass: Option[String] = None
+) extends Project {
+
+  def +(other: JsonProject): JsonProject =
+    JsonProject.monoid.orElse(this, other)
+
+  def writeTo(dir: os.Path): Unit = {
+    val config = WriterConfig.withIndentionStep(1)
+
+    Using(os.write.outputStream(dir / "export.json")) {
+      outputStream =>
+        writeToStream(
+          this,
+          outputStream,
+          config
+        )
+    }
+  }
+}
+
+case class ExportResolverFormat(
+  name: String,
+  location: Option[String],
+  user: Option[String] = None,
+  password: Option[String] = None,
+  headers: Seq[(String, String)] = Nil
+)
+
+object ExportResolverFormat {
+  implicit lazy val jsonCodec: JsonValueCodec[ExportResolverFormat] =
+    JsonCodecMaker.make(CodecMakerConfig
+      .withTransientNone(false)
+      .withTransientEmpty(false))
+  implicit lazy val StringSeqCodec: JsonValueCodec[Seq[String]] = JsonCodecMaker.make
+}
+
+object JsonProject {
+  implicit val monoid: ConfigMonoid[JsonProject]           = ConfigMonoid.derive
+  implicit lazy val jsonCodec: JsonValueCodec[JsonProject] = JsonCodecMaker.make
+}
+
+case class ExportDependencyFormat(groupId: String, artifactId: ArtifactId, version: String)
```

Comment:
Does that imply that dependencies will end up like this in the output JSON?
```json
{
  "groupId": "org.something",
  "artifactId": {
    "name": "foo",
    "fullName": "foo_3"
  },
  "version": "X.Y.Z"
}
```

That's way too heavyweight IMO. Dependencies have a concise string representation, meant to be used in that kind of case: `"org.something::foo:X.Y.Z"`, that can be parsed back to a non-ambiguous dependency.

Reply:
We should have dependencies as `dependency.DependencyLike` instances (or one of its aliases). It has a [`render`](https://github.com/coursier/dependency/blob/3a77788ad257ddf81a9d578b032e9c420c512bd4/dependency/src/main/scala/dependency/DependencyLike.scala#L36) method meant to provide that concise string.

Reply:
I would say the format we should use here depends on the particular use cases that we want to support. On the one hand encoding a dependency as a single string is more space-efficient but if we expect other tools to be built on the base of scala-cli, we would force them to reimplement the logic of parsing this dependency format, which might get slightly more complicated in the future (e.g. there have been some suggestions about implementing some heuristics to allow users to always write just `:` instead of either `:` or `::`). Also if this JSON grows too large it might get hard to read for humans and one of the motivations of this feature was also to let users clearly see the entire configuration of their project in case using directives are split between multiple files (which might be a case e.g. if you need different settings for different targets)